pkg/api/networkservice/mechanisms/wireguard/helpers.go

```diff
@@ -23,80 +23,101 @@ import (
-	SrcPort() int
-	// SrcPort - Destination interface listening port
-	DstPort() int
-}
-
-type mechanism struct {
+// Mechanism is a wireguard mechanism helper
+type Mechanism struct {
 	*networkservice.Mechanism
 }
 
 // ToMechanism - convert unified mechanism to useful wrapper
-func ToMechanism(m *networkservice.Mechanism) Mechanism {
+func ToMechanism(m *networkservice.Mechanism) *Mechanism {
 	if m.Type == MECHANISM {
 		if m.GetParameters() == nil {
 			m.Parameters = map[string]string{}
 		}
-		return &mechanism{
+		return &Mechanism{
 			m,
 		}
 	}
 	return nil
 }
 
-func (m *mechanism) SrcIP() net.IP {
+// SrcIP returns source ip
+func (m *Mechanism) SrcIP() net.IP {
 	return net.ParseIP(m.GetParameters()[SrcIP])
 }
 
-func (m *mechanism) DstIP() net.IP {
+// SetSrcIP sets source ip
+func (m *Mechanism) SetSrcIP(ip net.IP) {
+	m.GetParameters()[SrcIP] = ip.String()
+}
+
+// DstIP returns destination ip
+func (m *Mechanism) DstIP() net.IP {
 	return net.ParseIP(m.GetParameters()[DstIP])
 }
 
+// SetDstIP sets destination ip
+func (m *Mechanism) SetDstIP(ip net.IP) {
+	m.GetParameters()[DstIP] = ip.String()
+}
+
 // SrcPublicKey returns the SrcPublicKey parameter of the Mechanism
-func (m *mechanism) SrcPublicKey() string {
+func (m *Mechanism) SrcPublicKey() string {
 	return m.GetParameters()[SrcPublicKey]
 }
 
+// SetSrcPublicKey sets new source public key
+func (m *Mechanism) SetSrcPublicKey(key string) {
+	m.GetParameters()[SrcPublicKey] = key
+}
+
 // DstPublicKey returns the DstPublicKey parameter of the Mechanism
-func (m *mechanism) DstPublicKey() string {
+func (m *Mechanism) DstPublicKey() string {
 	return m.GetParameters()[DstPublicKey]
 }
 
+// SetDstPublicKey sets new destination public key
+func (m *Mechanism) SetDstPublicKey(key string) {
+	m.GetParameters()[DstPublicKey] = key
+}
+
 // SrcPort - Source interface listening port
-func (m *mechanism) SrcPort() int {
-	return atoi(m.GetParameters()[SrcPort])
+func (m *Mechanism) SrcPort() uint16 {
+	return atou16(m.GetParameters()[SrcPort])
+}
+
+// SetSrcPort sets source udp port
+func (m *Mechanism) SetSrcPort(port uint16) {
+	m.GetParameters()[SrcPort] = strconv.FormatUint(uint64(port), 16)
 }
 
 // DstPort - Destination interface listening port
-func (m *mechanism) DstPort() int {
-	return atoi(m.GetParameters()[DstPort])
+func (m *Mechanism) DstPort() uint16 {
+	return atou16(m.GetParameters()[DstPort])
+}
+
+// SetDstPort sets destination udp port
+func (m *Mechanism) SetDstPort(port uint16) {
+	m.GetParameters()[DstPort] = strconv.FormatUint(uint64(port), 16)
 }
 
 // GetPort - returns unique port by connection ID for wireguard connection
-func GetPort(connID string) string {
-	id, err := strconv.ParseUint(connID, 16, 0)
+// Since the connId has a random value, use only the first byte to limit port value
+func GetPort(connID string) uint16 {
+	if len(connID) < 2 {
+		return BasePort
+	}
+	id, err := strconv.ParseUint(connID[:2], 16, 16)
```

Comment:
I'm not sure that it is safe, actually, we cannot guarantee that `connID[:2]` is valid uint

Reply:
if it is not valid we will get an error and BasePort will be used..

Reply:
It is not a good solution. Two requests for two different clients will broke forwarder.
I think we should keep API for wireguard mechanism simple (just get and put port) and delegate management of pors to forwarders which support wireguard.

Reply:
Are we deriving port from the Connection.Id ?  We don't actually know for sure anything *about* the Connection Id... so tying things to its structure here doesn't seem like a good idea.